src/tools/perf/uct_tests.cc

```diff
@@ -440,6 +442,7 @@ class uct_perf_test_runner {
                     progress_responder();
                     if (UCS_CIRCULAR_COMPARE8(sn, >, (psn_t)(send_sn + (fc_window / 2)))) {
                         /* Send ACK every half-window */
+                        wait_for_window(send_window);
```

Comment:
we have to check window before send_b, so send_b would not exceed number of outstanding messages. if we check after send_b, it would be earlier than really needed.